### PR TITLE
support Using Apple Metal Performance Shaders

### PR DESCRIPTION
**Pull request by subtleGradient**
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Great! A PR on pyannote repo would be lovely!</p>&mdash; @hbredin@sigmoid.social (@hbredin) <a href="https://twitter.com/hbredin/status/1602193252882563072?ref_src=twsrc%5Etfw">December 12, 2022</a></blockquote>

# 🫡

---

# TESTS

I dunno how to add real unit tests. But no tests would be lame. So… here ya go :D

```py
import torch
import pyannote.audio.pipelines.utils.getter; import importlib; importlib.reload(pyannote.audio.pipelines.utils.getter)


assert pyannote.audio.pipelines.utils.getter.get_devices(needs=None) == [torch.device('mps'), torch.device("cpu")]

assert pyannote.audio.pipelines.utils.getter.get_devices(needs=1) == [torch.device('mps')]
assert pyannote.audio.pipelines.utils.getter.get_devices(needs=2) == [torch.device('mps'), torch.device("cpu")]
assert pyannote.audio.pipelines.utils.getter.get_devices(needs=3) == [torch.device('mps'), torch.device("cpu"), torch.device("mps")]
assert pyannote.audio.pipelines.utils.getter.get_devices(needs=4) == [torch.device('mps'), torch.device("cpu"), torch.device("mps"), torch.device("cpu")]
assert pyannote.audio.pipelines.utils.getter.get_devices(needs=5) == [torch.device('mps'), torch.device("cpu"), torch.device("mps"), torch.device("cpu"), torch.device("mps")]


try:
    pyannote.audio.pipelines.utils.getter.get_devices(needs=-1)
    error = None    
except ValueError as err:
    error = err
assert error is not None

try:
    pyannote.audio.pipelines.utils.getter.get_devices(needs=0)
    error = None    
except ValueError as err:
    error = err
assert error is not None
```

### CODE REVIEW

**Comment by hbredin** [comment]
Thanks for the PR!

On 2nd thought, I think it would be better to allow the user to choose the device(s) explicitely when instantiating pipelines, something like `pipeline.to(device)` (same API as `torch.nn.Module.to`).

Related issues/discussions: #1108 #1155 #1059 

cc @mbui0327 (who mentionned a possible PR doing just this a few weeks ago...)

**Comment by subtleGradient** [comment]
Yes! I would love that consistency with other stuff. Feel free to close this PR if that's the direction 😎👍

**Comment by hbredin** [comment]
Closing as [this new PR](#1204) adds a `pipeline.to(device)` API that should work with any `torch.device`.

Would you mind testing it?
